frontend/src/routes/manage/shared.tsx

```diff
@@ -0,0 +1,575 @@
+import { Card, match, useColorScheme } from "@opencast/appkit";
+import { useState, useRef, useEffect, ReactNode } from "react";
+import { ParseKeys } from "i18next";
+import { useTranslation } from "react-i18next";
+import {
+    LuArrowDownNarrowWide,
+    LuArrowUpWideNarrow,
+    LuChevronLeft,
+    LuChevronRight,
+} from "react-icons/lu";
+
+import FirstPage from "../../icons/first-page.svg";
+import LastPage from "../../icons/last-page.svg";
+import { ManageRoute } from ".";
+import { COLORS } from "../../color";
+import { PageTitle } from "../../layout/header/ui";
+import { Breadcrumbs } from "../../ui/Breadcrumbs";
+import { Link } from "../../router";
+import { EventConnection, EventRow } from "./Video";
+import { SeriesConnection, SeriesRow, Entry } from "./Series";
+import { VideosSortColumn } from "./Video/__generated__/VideoManageQuery.graphql";
+import { SeriesSortColumn } from "./Series/__generated__/SeriesManageQuery.graphql";
+
+
+type ItemVars = {
+    order: {
+        column: SortColumn;
+        direction: SortDirection;
+    };
+    limit: number;
+    offset: number;
+};
+
+type SharedProps = {
+    connection: EventConnection | SeriesConnection;
+    vars: ItemVars;
+    additionalColumns?: ColumnProps[];
+};
+
+type ManageItemProps = SharedProps & {
+    titleKey: ParseKeys;
+}
+
+const LIMIT = 15;
+
+export const ManageItems: React.FC<ManageItemProps> = ({
+    connection,
+    vars,
+    titleKey,
+    additionalColumns,
+}) => {
+    const { t } = useTranslation();
+
+    const totalCount = connection.totalCount;
+    const limit = vars.limit ?? 15;
+    const pageParam = new URLSearchParams(document.location.search).get("page");
+    const page = pageParam ? parseInt(pageParam, 10) : 1;
+
+    useEffect(() => {
+        const maxPage = Math.max(Math.ceil(totalCount / limit), 1);
+
+        if (page > maxPage) {
+            window.location.href = `?page=${maxPage}`;
+        } else if (page < 1) {
+            window.location.href = "?page=1";
+        }
+    }, [page, totalCount, limit]);
```

Comment:
Setting `window.location` is a hard reload. Have you tried using `const { router } = useRouter();` and `router.replace(...)`.

Reply:
The first condition (`page > maxPage`) doesn't work anyway when the backend doesn't clamp the offset, so I removed that. We can't do the clamping in frontend either, since we'd have to know the total count before sending the query.
If someone decides to fiddle around with the page parameter and uses a value larger than the max possible page, then we'll display no items.